quinn-proto/src/connection.rs

```diff
@@ -1914,10 +1914,12 @@ impl Connection {
 
         let space = self.spaces[space_id as usize].as_mut().unwrap();
         let mut padded = false;
-        if let Header::Initial { .. } = header {
-            if buf.len() < MIN_INITIAL_SIZE - AEAD_TAG_SIZE {
-                buf.resize(MIN_INITIAL_SIZE - AEAD_TAG_SIZE, 0);
-                padded = true;
+        if self.side.is_client() {
```

Comment:
We should probably invert the nesting of these conditions, since the check for initial space is false more often.